### PR TITLE
Fixed #31773 -- Fixed preserving output_field in ExpressionWrapper for combined expressions.

### PR DESCRIPTION
**Pull request by felixxm**
ticket-31773

Regression in df32fd42b84cc6dbba173201f244491b0d154a63.

### CODE REVIEW

File: django/db/models/expressions.py [hunk 1]
```diff
@@ -858,7 +858,10 @@ class ExpressionWrapper(Expression):
 
     def __init__(self, expression, output_field):
         super().__init__(output_field=output_field)
-        if getattr(expression, '_output_field_or_none', True) is None:
+        if (
+            isinstance(expression, Expression) and
+            getattr(expression, '_output_field_or_none', None) is None
+        ):
```
Comment (felixxm) [inline django/db/models/expressions.py:864]: I haven't found a better way to distinct between `expression`\`s that don't have `_output_field_or_none` (e.g. `Q`) and `expression`\`s that have `_output_field_or_none` but it raises an `AttributeError` (e.g. `CombinedExpression` with `F`).

**Comment by felixxm** [comment]
@charettes Thanks :+1:

**Comment by felixxm** [comment]
buildbot, test on oracle.

**Comment by carltongibson** [comment]
Release notes etc?

**Comment by felixxm** [comment]
It's a RB for Django 3.1 so we don't need them.

**Comment by carltongibson** [comment]
Don't worry... Ah, yes, you beat me. 3.1. Thanks.

**Pull request merged**